compute/instances/create-start-instance/create_instance_test.go

```diff
@@ -199,3 +199,100 @@ func TestComputeCreateInstanceSnippets(t *testing.T) {
 		t.Errorf("deleteDisk got err: %v", err)
 	}
 }
+
+func TestComputeBulkCreateInstanceSnippets(t *testing.T) {
+	//ctx := context.Background()
+	//var r *rand.Rand = rand.New(
+	//	rand.NewSource(time.Now().UnixNano()))
+	//tc := testutil.SystemTest(t)
+	//zone := "europe-central2-b"
+	//instanceName := fmt.Sprintf("test-instance-%v-%v", time.Now().Format("01-02-2006"), r.Int())
+	//bootDiskName := fmt.Sprintf("test-disk-%v-%v", time.Now().Format("01-02-2006"), r.Int())
+	//diskName2 := fmt.Sprintf("test-disk-%v-%v", time.Now().Format("01-02-2006"), r.Int())
+	//networkName := "global/networks/default"
+	//subnetworkName := "regions/europe-central2/subnetworks/default"
+	//expectedResult := "Instance created"
+
+	/////
+	ctx := context.Background()
+
+	instanceTemplatesClient, err := compute.NewInstanceTemplatesRESTClient(ctx)
+	if err != nil {
+		t.Fatalf("NewInstanceTemplatesRESTClient: %v", err)
+	}
+	defer instanceTemplatesClient.Close()
+
+	var seededRand *rand.Rand = rand.New(
+		rand.NewSource(time.Now().UnixNano()))
+	tc := testutil.SystemTest(t)
+	zone := "europe-central2-b"
+	instanceTemplateName := "test-instance-template" + fmt.Sprint(seededRand.Int())
+	machineType := "n1-standard-1"
+	sourceImage := "projects/debian-cloud/global/images/family/debian-10"
+	networkName := "global/networks/default"
+
+	insertTemplateReq := &computepb.InsertInstanceTemplateRequest{
+		Project: tc.ProjectID,
+		InstanceTemplateResource: &computepb.InstanceTemplate{
+			Name: &instanceTemplateName,
+			Properties: &computepb.InstanceProperties{
+				MachineType: proto.String(machineType),
+				Disks: []*computepb.AttachedDisk{
+					{
+						InitializeParams: &computepb.AttachedDiskInitializeParams{
+							DiskSizeGb:  proto.Int64(10),
+							SourceImage: proto.String(sourceImage),
+						},
+						AutoDelete: proto.Bool(true),
+						Boot:       proto.Bool(true),
+						Type:       proto.String(computepb.AttachedDisk_PERSISTENT.String()),
+					},
+				},
+				NetworkInterfaces: []*computepb.NetworkInterface{
+					{
+						Name: proto.String(networkName),
+					},
+				},
+			},
+		},
+	}
+
+	op, err := instanceTemplatesClient.Insert(ctx, insertTemplateReq)
+	if err != nil {
+		t.Fatalf("unable to create instance template: %v", err)
+	}
+
+	if err = op.Wait(ctx); err != nil {
+		t.Errorf("unable to wait for the operation: %v", err)
+	}
+
+	namePattern := "i-#-" + fmt.Sprint(seededRand.Int())[0:5]
+	buf := &bytes.Buffer{}
+
+	_, err = createFiveInstances(buf, tc.ProjectID, zone, instanceTemplateName, namePattern)
+	if err != nil {
+		t.Errorf("createFiveInstances got err: %v", err)
+	}
+
+	for i := 1; i <= 5; i++ {
```

Comment:
the `createFiveInstances` function returns the Instance objects. we could just iterate over that list and delete those instances directly. that avoids the need for the string replacement in the pattern.